woocommerce/woocommerce-ios
# Hub Menu: Unify navigation to use navigation destination

Closes: #13677 
⚠️ This PR includes refactoring to navigation of the Hub Menu, please smoke test the navigation for every item to ensure no regression is made.



## Description

Currently, we're using different ways to handle navigation in the hub menu: with raw IDs for taps on menu items, `NavigationPath` for payments' deep link, and `isPresented` bool value for coupons and review details deep links. 

Another problem is that we were adding an overlay of an invisible navigation link to handle navigation for taps on menu items, which messes with the voiceover feature for these items.

This PR fixes both issues above by:
- Removing the invisible navigation link overlay on top of each menu item.
- Supporting navigation with `NavigationPath` for all menu items except for Google Ads, as well as deep links for coupons and review details.
- Making `ProductReviewFromNoteParcel` conform to `Hashable` and `Equatable` to support handing review details with `NavigationPath`.


## Steps to reproduce

1. Testing voice over
- Turn on the voiceover feature in your physical device (Settings app > Accessibility > Voice Over)
- Build the app to the device and navigate to the Menu tab.
- Double-tap on each item on the menu, confirm that the correct screen is displayed for each item.

2. Testing navigation
- Turn off the voiceover feature for easier navigation.
- Build the app and navigate to the Menu tab.
- Tap on each item on the menu, confirm that the correct screen is displayed for each item.

Deep link for reviews
- Build the app to a physical device, log in to a test store, and put the app in the background.
- Add a new review for your test store.
- Confirm that a push notification is sent to the device.
- Tap the push notification, confirm that the app is navigated to the Reviews tab and the correct review detail screen is displayed.

📝 Note: due to a regression issue [p1724311685690509-slack-C03L1NF1EA3], the deep link for coupons is not working at the time of writing. Depending on whether that issue is fixed and merged, we need to test to ensure this works fine:

- Log in to a store without any coupons.
- Navigate to Orders tab and select + to add a new order.
- Select any product then tap Add coupons.
- The empty coupon list presented would have a CTA to the coupon screen. Tap that CTA.
- Confirm that the app is navigated to the Menu tab and the coupon list is displayed.

## Tests ran by @itsmeichigo 
- On simulator iOS 18.1 (iPhone 15 and iPad 10th generation): navigation on every item on the hub menu screen, deep link for coupons from order creation.
- On physical devices (iPhone 12 Pro iOS 17.5.1 & iPad OS 17.6): navigation on every item on the hub menu screen with & without voiceover, deep link for coupons from order creation, review notifications, collect payments via Spotlight, privacy settings.

## Screenshots

The UI should look the same as before.

<img src="https://github.com/user-attachments/assets/e4c7de8a-fff7-41f8-99ef-22c47151f03a" width=320 />


---

## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:
